packages/enzyme/src/ShallowWrapper.js

```diff
@@ -243,6 +245,76 @@ function privateSetChildContext(adapter, wrapper, instance, renderedNode, getChi
   }
 }
 
+/**
+ * Recursively dive()s every custom component in a wrapper until
```

Comment:
i'm not sure this is viable - a wrapping component could, for example, wrap in a div. if we don't want to allow that, we need to explicitly check for it in both mount and shallow, and throw.

Reply:
This implementation _should_ support non-custom components too. It iterates the entire component tree (including host components) and just `dive()`s on custom ones. But to be absolutely sure I'll throw some `<div />`s into the tests.

Reply:
`.dive()` explicitly throws on divs, is what i mean :-)

Reply:
I added some extra `<div />`/`<span />`s to the tests and can verify this still works. It works because this method uses the adapter to determine if a component can be `dive()`ed before diving it. If it can't be `dive()`ed it continues recursing to the next level of children. So, it _does_ iterate the whole tree (until it finds its target) but it only `dive()`s components that can be `dive()`ed.